src/main/java/com/nike/cerberus/operation/core/CreateVpcOperation.java

```diff
@@ -0,0 +1,107 @@
+import com.nike.cerberus.command.core.CreateVpcCommand;
+import com.nike.cerberus.domain.EnvironmentMetadata;
+import com.nike.cerberus.domain.cloudformation.VpcParameters;
+import com.nike.cerberus.domain.environment.StackName;
+import com.nike.cerberus.operation.Operation;
+import com.nike.cerberus.service.CloudFormationService;
+import com.nike.cerberus.service.Ec2Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.List;
+import java.util.Map;
+
+import static com.nike.cerberus.ConfigConstants.MINIMUM_AZS;
+import static com.nike.cerberus.module.CerberusModule.CF_OBJECT_MAPPER;
+
+/**
+ * Creates the base components via CloudFormation used by all of Cerberus.
+ */
+public class CreateVpcOperation implements Operation<CreateVpcCommand> {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final EnvironmentMetadata environmentMetadata;
+
+    private final CloudFormationService cloudFormationService;
+
+    private final Ec2Service ec2Service;
+
+    private final ObjectMapper cloudFormationObjectMapper;
+
+    @Inject
+    public CreateVpcOperation(final EnvironmentMetadata environmentMetadata,
+                              final CloudFormationService cloudFormationService,
+                              final Ec2Service ec2Service,
+                              @Named(CF_OBJECT_MAPPER) final ObjectMapper cloudformationObjectMapper) {
+        this.environmentMetadata = environmentMetadata;
+        this.cloudFormationService = cloudFormationService;
+        this.ec2Service = ec2Service;
+        this.cloudFormationObjectMapper = cloudformationObjectMapper;
+    }
+
+    @Override
+    public void run(final CreateVpcCommand command) {
+        final String environmentName = environmentMetadata.getName();
+        final Map<Integer, String> azByIdentifier = mapAvailabilityZones();
+
+        final VpcParameters vpcParameters = new VpcParameters()
+                .setAz1(azByIdentifier.get(1))
+                .setAz2(azByIdentifier.get(2))
+                .setAz3(azByIdentifier.get(3));
+
+        vpcParameters.getTagParameters().setTagEmail(command.getTagsDelegate().getTagEmail());
+        vpcParameters.getTagParameters().setTagName(ConfigConstants.ENV_PREFIX + environmentName);
+        vpcParameters.getTagParameters().setTagCostcenter(command.getTagsDelegate().getTagCostcenter());
+
+        final TypeReference<Map<String, String>> typeReference = new TypeReference<Map<String, String>>() {};
+        final Map<String, String> parameters = cloudFormationObjectMapper.convertValue(vpcParameters, typeReference);
+
+        cloudFormationService.createStack(StackName.VPC.getFullName(environmentName),
+                parameters, ConfigConstants.VPC_STACK_TEMPLATE_PATH, true);    }
+
+    @Override
+    public boolean isRunnable(final CreateVpcCommand command) {
+        return true;
```

Comment:
Implement this will break if stack exists already

Reply:
- [] Go through all new commands that create new CF stacks with static names and make sure isRunnable returns false if the stack that the command wants to make already exists.